Mathlib/RingTheory/MvPolynomial/Basic.lean

```diff
@@ -169,6 +169,21 @@ instance [Finite σ] (N : ℕ) : Module.Finite R (restrictTotalDegree σ R N) :=
 
 end Degree
 
+section Algebra
+
+variable {R S σ : Type*} [CommSemiring R] [CommSemiring S] [Algebra R S]
+
+/-- If `S` is an `R`-algebra, then `MvPolynomial σ S` is a `MvPolynomial σ R` algebra. -/
+noncomputable instance : Algebra (MvPolynomial σ R) (MvPolynomial σ S) :=
```

Comment:
This gives a diamond for `Algebra (MvPolynomial σ R) (MvPolynomial σ (MvPolynomial σ R))`. Please include the warning in the docstring and maybe not have it as a instance by default.

Reply:
I made it a `local instance`.